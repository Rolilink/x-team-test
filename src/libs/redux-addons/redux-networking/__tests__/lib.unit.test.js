import withNetwork, { networkFactory, cases, selectors } from '../';


const { setSuccess, setFetching, setError } = cases;
const { getIsFetching, getError } = selectors();

describe('cases.setSuccess', () => {
  it('should set state to default state', () => {
    const state = {
      items: [],
      network: {
        isFetching: true,
        error: 'error',
      },
    };

    expect(setSuccess(state)).toMatchSnapshot();
  });
});

describe('cases.setFetching', () => {
  it('should set fetching to true and error to null', () => {
    const state = {
      items: [],
      network: {
        isFetching: false,
        error: 'error',
      },
    };

    expect(setFetching(state)).toMatchSnapshot();
  });
});

describe('cases.setError', () => {
  it('should set error state with the action payload and isFetching to false', () => {
    const state = {
      items: [],
      network: {
        isFetching: true,
        error: null,
      },
    };

    const action = { payload: 'error' };

    expect(setError(state, action)).toMatchSnapshot();
  });
});

describe('factory.networkFactory', () => {
  /* TODO: create passing properties scenario */
  it('should return default network state', () => {
    expect(networkFactory()).toMatchSnapshot();
  });
});

describe('enhancer.withNetwork', () => {
  it('should enhance received state to add network state', () => {
    const state = {
      items: [],
    };

    expect(withNetwork(state)).toMatchSnapshot();
  });

  it('should be able to initialize network state', () => {
    const state = {
      items: [],
    };
    const initialNetworkState = {
      isFetching: true,
      error: 'error',
    };

    expect(withNetwork(state, initialNetworkState)).toMatchSnapshot();
  });
});

describe('selectors.getIsFetching', () => {
  const state = {
    items: [],
    network: {
      isFetching: false,
      error: 'error',
    },
  };

  it('should return the isFetching value', () => {
    expect(getIsFetching(state)).toMatchSnapshot();
  });
});

describe('selectors.getError', () => {
  const state = {
    items: [],
    network: {
      isFetching: false,
      error: 'error',
    },
  };

  it('should return the error value', () => {
    expect(getError(state)).toMatchSnapshot();
  });
});
