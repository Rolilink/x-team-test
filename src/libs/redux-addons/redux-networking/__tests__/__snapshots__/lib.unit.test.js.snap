// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cases.setError should set error state with the action payload and isFetching to false 1`] = `
Object {
  "items": Array [],
  "network": Object {
    "error": "error",
    "isFetching": false,
  },
}
`;

exports[`cases.setFetching should set fetching to true and error to null 1`] = `
Object {
  "items": Array [],
  "network": Object {
    "error": null,
    "isFetching": true,
  },
}
`;

exports[`cases.setSuccess should set state to default state 1`] = `
Object {
  "items": Array [],
  "network": Object {
    "error": null,
    "isFetching": false,
  },
}
`;

exports[`enhancer.withNetwork should be able to initialize network state 1`] = `
Object {
  "items": Array [],
  "network": Object {
    "error": "error",
    "isFetching": true,
  },
}
`;

exports[`enhancer.withNetwork should enhance received state to add network state 1`] = `
Object {
  "items": Array [],
  "network": Object {
    "error": null,
    "isFetching": false,
  },
}
`;

exports[`factory.networkFactory should return default network state 1`] = `
Object {
  "error": null,
  "isFetching": false,
}
`;

exports[`selectors.getError should return the error value 1`] = `"error"`;

exports[`selectors.getIsFetching should return the isFetching value 1`] = `false`;

exports[`selectors.getNetworkState should return the network subtree of the state 1`] = `
Object {
  "error": "error",
  "isFetching": false,
}
`;
